### PR TITLE
Termius.Termius version 8.0.4

### LINKED ISSUE #114544: [Package Issue]: Termius [Termius.Termius] installation fails with hash mismatch
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installer hash mismatch.

### Brief description of your issue

Installation for Termius [Termius.Termius] Version 7.59.2 fails with a "Installer hash does not match" error.

### Steps to reproduce

Running `winget install Termius.Termius`

### Actual behavior

Installation fails with `Installer hash does not match; to override this check use --force`

### Expected behavior

Successfull instalation of Termius package, version 7.59.2.

### Environment

```raw
Windows Package Manager v1.2.10691
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.1992
Package: Microsoft.DesktopAppInstaller v1.17.10691.0
```

### Screenshots and Logs

_No response_

### PR DESCRIPTION
**Pull request by borque**
Closes #114544
Replaces PR #114707 (vanity-url-installer was updated before pipelines had processed it and after it was reviewed by @ItzLevvie )
Related to PR #114716

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/winget-pkgs/pull/115322&drop=dogfoodAlpha

### CODE REVIEW

**Comment by borque** [comment]
@stephengillie or @ItzLevvie could you please start the pipelines and review this before the installer is updates again? I don't have enoug privileges to execute /azp run or something like that!

File: manifests/t/Termius/Termius/8.0.4/Termius.Termius.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Protocols:
+- http
+- ssh
+- telnet
+FileExtensions:
+- pem
+- ppk
+AppsAndFeaturesEntries:
+- DisplayName: Termius 8.0.4
+- DisplayVersion: 8.0.4
```
Comment (mdanish-kh) [inline manifests/t/Termius/Termius/8.0.4/Termius.Termius.installer.yaml:21]:
```suggestion
```
These are not needed as PackageName, Publisher are a good match for correlation already

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

**Review by mdanish-kh — APPROVED** [review]

**Comment by borque** [comment]
@mdanish-kh : I thought, I finally got it with the ARP-entries. But I was wrong. I do not understand when they are needed and when not. Is there any documentation for this?

**Comment by mdanish-kh** [comment]
@borque I'm writing this off the top of my head, so don't hesitate to ask if something doesn't make sense

ARP Entries can be used when we may need a different PackageName or PackageVersion in the manifest to show to the user that is different from what is put in the Registry/Apps&Features. If the PackageName, PackageVersion already match the corresponding values displayed in `appwiz.cpl` / Apps&Features, then there is probably not a need for these entries.

One common use case for ARP entries is to use it when a Package's marketing version (that publisher uses to market on it's website/homepage) is different from the one put into the Registry. We put the marketing version in place of PackageVersion and then use AppsAndFeatureEntries: DisplayVersion for explicit matching with the registry values. These ARP entries, in particular `DisplayVersion` should only be used when required as it causes additional version mapping for the client and it then requires that every new manifest must include the `DisplayVersion` property to prevent infinite upgrades for the user.

**Comment by borque** [comment]
@mdanish-kh thank you very much for the fast and good explanation! At least my understanding was not co pletly wrong: The packages name is simply „Terminus“ but the ARP-entry for name is „Terminus 8.0.4“ and so I thought it was necessary because of this difference.

**Comment by mdanish-kh** [comment]
@borque Ah so that's the other thing; WinGet does a fuzzy match for PackageName. You can ignore the version bit that comes after. The match needs to be "good enough" and you won't have any correlation issues.

Including `AppsAndFeatures: DisplayName: Terminus 8.0.4` would require contributors to update this field manually (as these fields are not edited by manifest creators for very obvious reasons as they can be any new random value) for every new version. That's really not required for this package

**Comment by borque** [comment]
@mdanish-kh : okay, once again I think I got it. Well, I will see… It is great how you explain the problems. Thanks once again. May I bother you with two further questions?
1) Whether the names match is good enough for WinGet can only be a reasonable assumption by the contributor, or are there certain rules that need to be obeyed?
2) What about automatic updates by Winget-Bot (e.g. for vanity-url-installers): is he able to change and correct the ARP-section in manifests?

**Comment by mdanish-kh** [comment]
> 1. Whether the names match is good enough for WinGet can only be a reasonable assumption by the contributor, or are there certain rules that need to be obeyed?

It is generally advisable to exclude the parts that contain the version, release date or any other sort of metadata that is going to change in a newer release _if_ they appear at the end of the DisplayName. Take this https://github.com/microsoft/winget-pkgs/pull/87670/files as an example. The version was in between the DisplayName in Apps&Features so I had to put PackageName `AnyLogic Professional` as the user-friendly name that is shown when users do `winget search` / `winget show` / `winget install` for this manifest but use AppsAndFeaturesEntries for better correlation for the WinGet client.
 
> 2. What about automatic updates by Winget-Bot (e.g. for vanity-url-installers): is he able to change and correct the ARP-section in manifests?

Nope; We have to close those PRs and create a new one fixing the DisplayVersion. That is one reason we don't want more inclusions like these when they are not required. You can take a look at the list of these PRs with incorrect DisplayVersions (they usually get flagged with `Manifest-AppsAndFeaturesVersion-Error` when I think it detects a duplicate DisplayVersion)
https://github.com/microsoft/winget-pkgs/pulls?q=is%3Apr+label%3AManifest-AppsAndFeaturesVersion-Error+author%3Awingetbot+is%3Aclosed

**Pull request merged**